optaplanner-core/src/test/java/org/optaplanner/core/api/score/holder/ScoreHolderTest.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.optaplanner.core.api.score.holder;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.optaplanner.core.api.score.Score;
+
+import static org.junit.Assert.assertNotNull;
+
+public class ScoreHolderTest {
+
+    @Rule
+    public final ExpectedException thrown = ExpectedException.none();
+
+    @Test
+    public void illegalStateExceptionThrownWhenConstraintMatchNotEnabled() {
+        ScoreHolder scoreHolder = buildScoreHolder(false);
```

Comment:
I intentionally keep the same name that is used in JUnit examples. Examples (of any project) form a de facto convention and if you follow this convention, your code might be easier to read for newcomers. Just to explain my reasoning.

If you like "expectedException" better, I will happily change it and, of course, keep using that in the future for the sake of consistency.